node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentImpl.java

```diff
@@ -102,9 +103,9 @@
     public NodeAgentImpl(NodeAgentContextSupplier contextSupplier, NodeRepository nodeRepository,
                          Orchestrator orchestrator, DockerOperations dockerOperations, StorageMaintainer storageMaintainer,
                          FlagSource flagSource, Optional<CredentialsMaintainer> credentialsMaintainer,
-                         Optional<AclMaintainer> aclMaintainer, Optional<HealthChecker> healthChecker, Clock clock) {
+                         Optional<AclMaintainer> aclMaintainer, Optional<HealthChecker> healthChecker, Clock clock, ZoneApi zone) {
         this(contextSupplier, nodeRepository, orchestrator, dockerOperations, storageMaintainer, flagSource, credentialsMaintainer,
-                aclMaintainer, healthChecker, clock, DEFAULT_WARM_UP_DURATION);
+             aclMaintainer, healthChecker, clock, warmUpDurationForZone(zone));
```

Comment:
Consider just resolving this when needed via the `ZoneApi` in `NodeAgentContext`

Reply:
NodeAgentContextSupplier.nextContext() says `blocks until the next context is ready`and that sounded like something I wouldn't do in a constructor, I think I'll leave this for now at least

Reply:
No, I meant in the methods that actually use this variable (`startContainer()`, `updateContainerIfNeeded()` and `doConverge()`).

Another option is to pass the initial context to the constructor instead of just the zone, since `NodeAgentFactory` already has it https://github.com/vespa-engine/vespa/blob/44e2be1ea393b7556ab59e7f77a757d8d25be546/node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentFactory.java#L9

Reply:
Yes, I initially did resolving when needed, let's go back to it. Thanks, done